src/lib/structures/expr.ml

```diff
@@ -377,175 +377,380 @@ let print_binders =
       print_one fmt e;
       List.iter (fun e -> fprintf fmt ", %a" print_one e) l
 
-let rec print_silent fmt t =
-  let { f ; xs ; ty; bind; _ } = t in
-  match f, xs with
-  (* Formulas *)
-  | Sy.Form form, xs ->
-    begin
-      match form, xs, bind with
-      | Sy.F_Unit _, [f1; f2], _ ->
-        fprintf fmt "@[(%a /\\@ %a)@]" print_silent f1 print_silent f2
-
-      | Sy.F_Iff, [f1; f2], _ ->
-        fprintf fmt "@[(%a <->@ %a)@]" print_silent f1 print_silent f2
-
-      | Sy.F_Xor, [f1; f2], _ ->
-        fprintf fmt "@[(%a xor@ %a)@]" print_silent f1 print_silent f2
-
-      | Sy.F_Clause _, [f1; f2], _ ->
-        fprintf fmt "@[(%a \\/@ %a)@]" print_silent f1 print_silent f2
-
-      | Sy.F_Lemma, [], B_lemma { user_trs ; main ; name ; binders; _ } ->
-        if get_verbose () then
-          fprintf fmt "(lemma: %s forall %a[%a].@  %a)"
-            name
-            print_binders binders
-            print_triggers user_trs
-            print_silent main
-        else
-          fprintf fmt "(lem %s)" name
+(* let print_list_sep sep pp fmt =
+ *   Format.pp_print_list ~pp_sep:(fun fmt _ -> Format.fprintf fmt sep) pp fmt
+ *
+ * let print_list pp fmt = print_list_sep "," pp fmt *)
+
+module SmtPrinter = struct
+
+  let rec print_formula fmt form xs bind =
+    match form, xs, bind with
+    | Sy.F_Unit _, [f1; f2], _ ->
+      fprintf fmt "@[(%a /\\@ %a)@]" print_silent f1 print_silent f2
```

Comment:
these cases do not output correct smtlib2, either they should be fixed (e.g. conjunction is `(and foo bar)` not `(foo /\ bar)`), or at the very least, these cases should raise en exception.